lib/resty/mlcache.lua

```diff
@@ -716,7 +722,9 @@ local function run_callback(self, key, shm_key, data, ttl, neg_ttl,
     -- finished to run the callback
 
     if lerr == "timeout" then
-        local errmsg = "could not acquire callback lock: timeout"
+        local errmsg = "refreshing cache failed: a previous trial is still" ..
+                       " running and we timed out waiting for it to finish." ..
+                       " (key: " .. key .. ")"
```

Comment:
I do not like this change. Let's keep the original error and add the `key: ...` suffix, that would be enough.